src/ConcordiumNetSdk/Types/Expiry.cs

```diff
@@ -0,0 +1,127 @@
+using ConcordiumNetSdk.Helpers;
+
+namespace ConcordiumNetSdk.Types
+{
+    /// <summary>
+    /// Expiration time for a transaction.
+    /// </summary>
+    public class Expiry
+    {
+        public const int BytesLength = sizeof(UInt64);
+
+        /// <summary>
+        /// Time at which the transaction expires.
+        /// </summary>
+        private readonly DateTimeOffset _timestamp;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Expiry"/> class.
+        /// </summary>
+        /// <param name="timestamp">Time at which the transaction expires.</param>
+        private Expiry(DateTimeOffset timestamp)
+        {
+            _timestamp = timestamp;
+        }
+
+        /// <summary>
+        /// Returns a new expiration time that is <c>minutes</c> ahead of the receiver object.
+        /// </summary>
+        /// <param name="minutes">Number of minutes later the resulting expiration times should be, relative to <c>this</c>.</param>
+        public Expiry AddMinutes(UInt64 minutes)
+        {
+            return Expiry.From(_timestamp.AddMinutes(minutes));
+        }
+
+        /// <summary>
+        /// Returns a new expiration time that is <c>seconds</c> ahead of the receiver object.
+        /// </summary>
+        /// <param name="minutes">Number of seconds later the resulting expiration times should be, relative to <c>this</c>.</param>
+        public Expiry AddSeconds(UInt64 seconds)
+        {
+            return Expiry.From(_timestamp.AddSeconds(seconds));
+        }
```

Comment:
This is not really how Expiry is used. Essentially you want to just construct an expiry "n seconds" or "n minutes" from now.

You don't want to construct something and then add to it.